File: stockmax/stockmax.cpp
/**
 * Code by longtqtqn
 */
#include <bits/stdc++.h>

using namespace std;

void solve() {
    int n;
    cin >> n;
    vector<int> prices(n);
    for (int &x : prices) {
        cin >> x;
    }

    int64_t curMax = prices[n - 1];
    int64_t rs = 0;
    for (int i = n - 2; i >= 0; --i) {
        if (prices[i] < curMax) {
            rs += curMax - prices[i];
        } else {
            curMax = prices[i];
        }
    }

    cout << rs << "\n";
}

int main() {
    int t;
    cin >> t;
    while(t--) {
        solve();
    }

}

File: stockmax/stockmax.js
/**
 * Code by longtqtqn
 */

process.stdin.resume();
process.stdin.setEncoding('utf-8');

_input = '';

process.stdin.on('data', function(inputStdin) {
    _input += inputStdin;
});

process.stdin.on('end', function() {
    main(_input);
});

function main(input) {
    var inputArr, 
        prices = [],
        n,
        curLine = 0;
    
    inputArr = input.split('\n');
    
    t = parseInt(inputArr[curLine++]);
    
    for (it = 0; it < t; ++it) {
        n = parseInt(inputArr[curLine++])
        prices = inputArr[curLine++].split(' ').map(function(n) {
            return parseInt(n, 10);
        })
        solve(n, prices);
    }
}

function solve(n, prices) {
    var curMax = prices[n - 1];
    var rs = 0;
    var i = n - 2;
    for (; i >= 0; --i) {
        if (prices[i] < curMax) {
            rs += curMax - prices[i];
        } else {
            curMax = prices[i];
        }
    }
    console.log(rs);
}
